pkg/policy/consumer.go

```diff
@@ -258,37 +245,34 @@ func (c *Consumable) AllowConsumerAndReverseLocked(cache *ConsumableCache, id Nu
 		}).Debug("Allowing reverse direction")
 		if _, ok := reverse.ReverseRules[c.ID]; !ok {
 			reverse.addToMaps(c.ID)
-			reverse.ReverseRules[c.ID] = NewConsumer(c.ID)
+			reverse.ReverseRules[c.ID] = true
 			return true
 		}
 	}
 	log.WithFields(logrus.Fields{
 		logfields.Identity + ".from": c.ID,
 		logfields.Identity + ".to":   id,
-	}).Warn("Allowed a consumer which can't be found in the reverse direction")
+	}).Warn("Allowed an ingress security identity which can't be found in the reverse direction")
 	return changed
 }
 
-// BanConsumerLocked removes the given consumer from the Consumable's consumers
-// map. Must be called with the Consumable mutex locked.
-func (c *Consumable) BanConsumerLocked(id NumericIdentity) {
-	if consumer, ok := c.Consumers[id.StringID()]; ok {
-		log.WithField("consumer", logfields.Repr(consumer)).Debug("Removing consumer")
-		delete(c.Consumers, id.StringID())
+// RemoveIngressIdentityLocked removes the given security identity from Consumable's
+// IngressIdentities map.
+// Must be called with the Consumable mutex locked.
+func (c *Consumable) RemoveIngressIdentityLocked(id NumericIdentity) {
+	if _, ok := c.IngressIdentities[id]; ok {
+		log.WithField(logfields.Identity, id).Debug("Removing ingress identity")
+		delete(c.IngressIdentities, id)
 
 		if c.wasLastRule(id) {
 			c.removeFromMaps(id)
 		}
-
-		if consumer.Reverse != nil {
-			c.deleteReverseRule(id, c.ID)
-		}
 	}
 }
 
 func (c *Consumable) Allows(id NumericIdentity) bool {
 	c.Mutex.RLock()
-	consumer := c.getConsumer(id)
+	identityAllowed := c.isIdentityAllowed(id)
 	c.Mutex.RUnlock()
-	return consumer != nil && consumer.Decision == api.Allowed
+	return identityAllowed != false
 }
```

Comment:
Would be cleaner to defer the RUnlock and then `return c.isIdentityAllowed(id)`

Reply:
Done.

Reply:
Defer adds overhead in small functions. Is is ok if it goes like this?
```golang
func (c *Consumable) Allows(id NumericIdentity) (isAllowed bool) {
 	c.Mutex.RLock()
 	isAllowed = c.getConsumer(id) != false
 	c.Mutex.RUnlock()
        return
}
```

Reply:
> Defer adds overhead in small functions. Is is ok if it goes like this?

I have no strong opinion on this, so I'll revert that change.

Reply:
Fixed in latest upload.